crates/astria-sequencer/src/service/mempool.rs

```diff
@@ -187,6 +187,17 @@ async fn handle_check_tx<S: StateReadExt + 'static>(
         };
     };
 
+    if mempool.check_invalid(tx_hash).await {
+        mempool.remove(tx_hash).await;
+        metrics::counter!(metrics_init::CHECK_TX_REMOVED_INVALID).increment(1);
+        return response::CheckTx {
+            code: AbciErrorCode::INVALID_PARAMETER.into(),
```

Comment:
Not sure if this is a good code - maybe we need a new one for this case?

Reply:
yeah, a code specifically for this would be good